packages/cinterop/src/androidTest/java/io/realm/test/sync/SyncEnumTests.kt

```diff
@@ -0,0 +1,153 @@
+import io.realm.internal.interop.sync.JsonErrorCode
+import io.realm.internal.interop.sync.MetadataMode
+import io.realm.internal.interop.sync.ProtocolClientErrorCode
+import io.realm.internal.interop.sync.ProtocolConnectionErrorCode
+import io.realm.internal.interop.sync.ProtocolSessionErrorCode
+import io.realm.internal.interop.sync.ServiceErrorCode
+import io.realm.internal.interop.sync.SyncErrorCodeCategory
+import org.junit.Test
+import kotlin.reflect.KClass
+import kotlin.test.BeforeTest
+import kotlin.test.assertEquals
+
+/**
+ * Test that sync related enum wrappers map all values, which is relevant when the Core API changes.
+ * This test is isolated to the JVM as Native doesn't have the reflection capabilities required
+ * to test this efficiently.
+ */
+class SyncEnumTests {
+
+    @BeforeTest
+    fun setup() {
+        System.loadLibrary("realmc")
+    }
+
+    @Test
+    fun appErrorCategory() {
+        checkEnum(realm_app_error_category_e::class) { nativeValue ->
+            AppErrorCategory.of(nativeValue)
+        }
+    }
+
+    @Test
+    fun authProvider() {
+        checkEnum(realm_auth_provider_e::class) { nativeValue ->
+            AuthProvider.of(nativeValue)
+        }
+    }
+
+    @Test
+    fun clientErrorCode() {
+        checkEnum(realm_app_errno_client_e::class) { nativeValue ->
+            ClientErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    @Test
+    fun coreUserState() {
+        checkEnum(realm_user_state_e::class) { nativeValue ->
+            CoreUserState.of(nativeValue)
+        }
+    }
+
+    @Test
+    fun metadataMode() {
+        checkEnum(realm_sync_client_metadata_mode_e::class) { nativeValue ->
+            MetadataMode.of(nativeValue)
+        }
+    }
+
+    @Test
+    fun protocolClientErrorCode() {
+        checkEnum(realm_sync_errno_client_e::class) { nativeValue ->
+            ProtocolClientErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    @Test
+    fun protocolConnectionErrorCode() {
+        checkEnum(realm_sync_errno_connection_e::class) { nativeValue ->
+            ProtocolConnectionErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    @Test
+    fun protocolSessionErrorCode() {
+        checkEnum(realm_sync_errno_session_e::class) { nativeValue ->
+            ProtocolSessionErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    @Test
+    fun serviceErrorCode() {
+        checkEnum(realm_app_errno_service_e::class) { nativeValue ->
+            ServiceErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    @Test
+    fun syncErrorCodeCategory() {
+        checkEnum(realm_sync_error_category_e::class) { nativeValue ->
+            SyncErrorCodeCategory.of(nativeValue)
+        }
+    }
+
+    @Test
+    fun jsonErrorCode() {
+        checkEnum(realm_app_errno_json_e::class) { nativeValue ->
+            JsonErrorCode.fromInt(nativeValue)
+        }
+    }
+
+    private inline fun <T : Any> checkEnum(enumClass: KClass<out Any>, mapNativeValue: (Int) -> T) {
```

Comment:
Always nice with a test that verifies that such an assisting assertion method is actually catching an error. Don't know if you can put in some invalid definition though 🤔

Reply:
It is possible, but the chance is fairly slim as you have to mess up both the common code and you usually see both "our" name and the autogenerated one side by side. It also looks like something that is very hard to check, except through some clever use of reflection, so It didn't seem worth it.